### PR TITLE
Implement signal trait for GLArea

### PR DESCRIPTION
**Pull request by mjkoo**
Looked into what was needed to get a simple example with GtkGLArea working, had to change the signal names and expose the appropriate trait similar to other widgets. Was able to set up opengl using a dummy invisible glutin window and draw to the widget after setting this up.

Can clean up my example and submit a PR to the examples repo if you're interested. Right now it just uses gl::ClearColor to paint the whole widget red, looking into getting a triangle up akin to the example at https://www.bassi.io/articles/2015/02/17/using-opengl-with-gtk/

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Thanks for the PR ! We would be totally interested in an OpenGL example.

cc @gkoz.

File: src/signal.rs [hunk 1]
```diff
@@ -55,6 +55,12 @@ use {
     WidgetHelpType,
 };
 
+#[cfg(gtk_3_16)]
+use {
+    gdk,
```
Comment (gkoz) [inline src/signal.rs]: This conditional import of `gdk` could be problematic. Since we're importing a lot of stuff from `gdk` above it's okay to import `GLContext` as well.

**Comment by gkoz** [comment]
You don't need to add a trait there, it would've been enough to make the methods public (apparently, I've made a poor job of reviewing this code when it was checked in).

File: src/signal.rs (hunk 1 above)
Comment (mjkoo) [inline src/signal.rs]: Sure, actually removing the trait as mentioned in your other comment meant I could remove both of those.

**Comment by mjkoo** [comment]
Thanks gkoz, I removed the trait. I'm relatively new to rust and was following the example for the calendar widget below GLArea in src/signal.rs, could I bother you for an ELI5 why it's required for one and not the other? Or is the CalendarSignals trait also superfluous?

**Comment by gkoz** [comment]
You're doing very good!
A trait is needed in case the class has subclasses. Take `WidgetSignals` for example: each widget needs to have those methods. When the class is a leaf in the hierarchy we usually don't add additional traits but just implement inherent methods. I guess the existence `CalendarSignals` contradicts that.

**Comment by mjkoo** [comment]
Thanks! That makes sense, I wasn't sure if the trait was useful in client code (subclassing widgets in apps?) being the reason for its inclusion, I grepped for `CalendarSignals` originally and didn't see another use from the library itself. Thanks for the explanation!
